src/Castle.Core/DynamicProxy/Generators/MethodWithInvocationGenerator.cs

```diff
@@ -34,20 +34,28 @@ internal class MethodWithInvocationGenerator : MethodGenerator
 		private readonly IInvocationCreationContributor contributor;
 		private readonly GetTargetExpressionDelegate getTargetExpression;
 		private readonly GetTargetExpressionDelegate getTargetTypeExpression;
-		private readonly Reference interceptors;
+		private readonly Expression interceptors;
 		private readonly Type invocation;
 
 		public MethodWithInvocationGenerator(MetaMethod method, Reference interceptors, Type invocation,
 		                                     GetTargetExpressionDelegate getTargetExpression,
 		                                     OverrideMethodDelegate createMethod, IInvocationCreationContributor contributor)
-			: this(method, interceptors, invocation, getTargetExpression, null, createMethod, contributor)
+			: this(method, interceptors.ToExpression(), invocation, getTargetExpression, null, createMethod, contributor)
 		{
 		}
 
 		public MethodWithInvocationGenerator(MetaMethod method, Reference interceptors, Type invocation,
 		                                     GetTargetExpressionDelegate getTargetExpression,
 		                                     GetTargetExpressionDelegate getTargetTypeExpression,
 		                                     OverrideMethodDelegate createMethod, IInvocationCreationContributor contributor)
+			: this(method, interceptors.ToExpression(), invocation, getTargetExpression, getTargetTypeExpression, createMethod, contributor)
+		{
+		}
+
+		public MethodWithInvocationGenerator(MetaMethod method, Expression interceptors, Type invocation,
+		                                     GetTargetExpressionDelegate getTargetExpression,
+		                                     GetTargetExpressionDelegate getTargetTypeExpression,
+		                                     OverrideMethodDelegate createMethod, IInvocationCreationContributor contributor)
```

Comment:
This new overload is identical to the preceding one except that `interceptors` is an `Expression`, not a `Reference`. Internally, this class needs an `Expression` anyway, and this ctor allows us to pass in a `NullExpression` (see review comment above).